doc/ansible/dev/developer_guide.md

```diff
@@ -0,0 +1,430 @@
+Run git init ...
+Initialized empty Git repository in /home/dymurray/go/src/github.com/dymurray/opsdk/foo-operator/.git/
+Run git init done
+
+$ cd foo-operator
+```
+
+Modify `roles/Foo/tasks/main.yml` with desired Ansible logic. For this example
+we will create and delete a namespace with the switch of a variable:
+```yaml
+---
+- name: set test namespace to {{ state }}
+  k8s:
+    api_version: v1
+    kind: Namespace
+    state: "{{ state }}"
+  ignore_errors: true
+```
+**note**: Setting `ignore_errors: true` is done so that deleting a nonexistent
+project doesn't error out.
+
+Modify `roles/Foo/defaults/main.yml` to set `state` to `present` by default.
+```yaml
+---
+state: present
+```
+
+Create an Ansible playbook `playbook.yaml` in the top-level directory which
+includes role `Foo`:
+```yaml
+---
+- hosts: localhost
+  roles:
+    - Foo
+```
+
+Run the playbook:
+```bash
+$ ansible-playbook playbook.yaml
+ [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
+
+
+PLAY [localhost] ***************************************************************************
+
+TASK [Gathering Facts] *********************************************************************
+ok: [localhost]
+
+Task [Foo : set test namespace to present]
+changed: [localhost]
+
+PLAY RECAP *********************************************************************************
+localhost                  : ok=2    changed=1    unreachable=0    failed=0
+
+```
+
+Check that the namespace was created:
+```bash
+$ kubectl get namespace
+NAME          STATUS    AGE
+default       Active    28d
+kube-public   Active    28d
+kube-system   Active    28d
+test          Active    3s
+```
+
+Rerun the playbook setting `state` to `absent`:
+```bash
+$ ansible-playbook playbook.yml --extra-vars state=absent
+ [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
+
+
+PLAY [localhost] ***************************************************************************
+
+TASK [Gathering Facts] *********************************************************************
+ok: [localhost]
+
+Task [Foo : set test namespace to absent]
+changed: [localhost]
+
+PLAY RECAP *********************************************************************************
+localhost                  : ok=2    changed=1    unreachable=0    failed=0
+
+```
+
+Check that the namespace was deleted:
+```bash
+$ kubectl get namespace
+NAME          STATUS    AGE
+default       Active    28d
+kube-public   Active    28d
+kube-system   Active    28d
+```
+## Using Ansible inside of an Operator
+Now that we have demonstrated using the k8s modules inside of Ansible, we want
+to trigger this Ansible logic when a custom resource changes. In the above
+example, we want to map a role to a specific Kubernetes resource that the
+operator will watch. This is called the Watches file.
```

Comment:
Since you've only described the problem so far ("we need to $x"), I'd introduce the watches file more explicitly as the solution. Perhaps something like: "This mapping is done in a file called `watches.yaml`".